skimage/transform/_ring_detector.pyx

```diff
@@ -0,0 +1,1184 @@
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#  
+################################################################################
+
+## EA NOTES
+# * EA20200724 : the opencv version shows better performace:
+#                * pre-processing takes x3 longer using hessian_matrix 
+#                  (skimage), and 
+#                * ring_detection takes about x1.2, showing higher 
+#                  sensitivity to choice of parameter (sigma and curv_thresh)
+# * EA20200724 : results seem more precise and less sensitive to choice of
+#                parameters when taking derivatives usind the
+#                filters.farid_v/h (or filters.scharr_v/h); however, these
+#                operators take x6  (x3) times longer to compute, w.r.t to 
+#                hessian_matrix, which uses np.gradient; this could probably
+#                be improved by programming them to allow the 2nd order
+#                derivatives directly.
+#
+
+## EA To consider 
+# * EA20200723 : store both principal curvatures and testing for  
+#                |Lpp|>|Lqq| for Lpp being the least principal curvature
+# * EA20200724 : allow the user to choose between np.gradient (skimage
+#                hessian_matrix), scharr, and farid operators; the latter 
+#                seem to offer more precise results, with less sensitivity to
+#                the choice of parameters. The former is faster.
+# * EA20200723 : add automated choice of curv_thresh (see notebook for HY)
+# * EA20200723 : verify results and efficiency are not significantly affected
+#                the transition from opencv to skimage (Hessian calculation)
+# * EA20200723 : explore the effect of separating the ridge detection function
+#                entirely; it may not affect the %%timeit results.
+
+## ancient TODOs
+
+# TODO: cast indices to unsigned int
+# TODO: add a test to verify rij <-> coords conversion works as expected
+# (assersion)
+# TODO: add edge detection version
+
+from __future__ import division 
+from libc.math cimport sqrt, copysign, cos, abs, fabs, ceil, exp, log2
+from libc.math cimport M_PI as pi
+
+from cpython cimport bool
+
+#from collections import defaultdict
+
+import numpy as np
+cimport numpy as np
+
+#from skimage.feature import hessian_matrix, hessian_matrix_eigvals
+from ..feature import hessian_matrix, hessian_matrix_eigvals
+
+from scipy import optimize
+
+import cython
+cimport cython
+
+DTYPE = np.float32
+ctypedef np.float32_t DTYPE_t
+
+RIJ = np.uint32
+ctypedef np.uint32_t RIJ_t
+
+INDX = np.int
+ctypedef Py_ssize_t INDX_t
```

Comment:
This seems incorrect on 64bit windows.  Maybe just use `IDX = np.intp` and `ctypedef npy_intp INDX_T`.  Those don't have to match, bt it is what numpy uses (I am not sure cython uses it, but...)